fineract-provider/src/main/java/org/apache/fineract/portfolio/address/service/AddressWritePlatformServiceImpl.java

```diff
@@ -0,0 +1,317 @@
+
+		final Address add = Address.fromJson(command, stateIdobj, countryIdObj);
+		this.addressRepository.save(add);
+		final Long addressid = add.getId();
+		final Address addobj = this.addressRepository.getOne(addressid);
+
+		final Client client = this.clientRepository.getOne(clientId);
+
+		final ClientAddress clientAddressobj = ClientAddress.fromJson(command, client, addobj, addressTypeIdObj);
+		this.clientAddressRepository.save(clientAddressobj);
+
+		return new CommandProcessingResultBuilder().withCommandId(command.commandId())
+				.withEntityId(clientAddressobj.getId()).build();
+	}
+
+	@Override
+	public CommandProcessingResult addNewClientAddress(final Client client, final JsonCommand command) {
+		CodeValue stateIdobj = null;
+		CodeValue countryIdObj = null;
+		long stateId;
+		long countryId;
+
+		final JsonArray addressArray = command.arrayOfParameterNamed("address");
+
+		for (int i = 0; i < addressArray.size(); i++) {
+			final JsonObject jsonObject = addressArray.get(i).getAsJsonObject();
+			this.fromApiJsonDeserializer.validateForCreate(jsonObject.toString(), true);
+
+			if (command.longValueOfParameterNamed("state_province_id") != null) {
+				stateId = command.longValueOfParameterNamed("state_province_id");
+				stateIdobj = this.codeValueRepository.getOne(stateId);
+			}
+
+			if (command.longValueOfParameterNamed("country_id") != null) {
+				countryId = command.longValueOfParameterNamed("country_id");
+				countryIdObj = this.codeValueRepository.getOne(countryId);
+			}
+
+			final long addressTypeId = jsonObject.get("addressTypeId").getAsLong();
+			final CodeValue addressTypeIdObj = this.codeValueRepository.getOne(addressTypeId);
+
+			final Address add = Address.fromJsonObject(jsonObject, stateIdobj, countryIdObj);
+			this.addressRepository.save(add);
+			final Long addressid = add.getId();
+			final Address addobj = this.addressRepository.getOne(addressid);
+
+			final ClientAddress clientAddressobj = ClientAddress.fromJson(command, client, addobj, addressTypeIdObj);
+			this.clientAddressRepository.save(clientAddressobj);
+
+		}
+		final long typ = 1;
+		return new CommandProcessingResultBuilder().withCommandId(command.commandId()).withEntityId(typ).build();
+	}
+
+	@Override
+	public CommandProcessingResult updateClientAddress(final Long clientId, final Long addressTypeId,
+			final Boolean status, final JsonCommand command) {
+		this.context.authenticatedUser();
+
+		long stateId;
+
+		long countryId;
+
+		CodeValue stateIdobj;
+
+		CodeValue countryIdObj;
+
+		boolean is_address_update = false;
+
+		boolean is_address_config_update = false;
+
+		final CodeValue addresstyp = this.codeValueRepository.getOne(addressTypeId);
+		
+		
+		System.out.println("clientId is"+ clientId);
+		System.out.println("addresstyp:" +addresstyp);
+		System.out.println("status:" +status);
+
+		final ClientAddress clientAddressObj = this.clientAddressRepositoryWrapper
```

Comment:
Why this method is required to pass status? Any how you are always retrieving active client address right?